Class/Practise 5.2/include/MegaDataPool.hpp

```diff
@@ -0,0 +1,33 @@
+#pragma once
+
+#include <vector>
+#include <memory>
+#include <mutex>
+#include "MegaData.hpp"
+
+class MegaDataPool {
+public:
+    static MegaDataPool& getInstance(size_t poolSize = 10) {
+        static MegaDataPool instance(poolSize);
```

Comment:
Minor: In some cases we may be interested to allocate that big object on heap(e.g. we can use unique_ptr here as a storage) especially if this singleton is rarely used. But of course it is an extra optimisation that can be considered only if some performance gap is observed.